timeoutMiddleware.go

```diff
@@ -0,0 +1,40 @@
+package gin
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	contractshttp "github.com/goravel/framework/contracts/http"
+	"github.com/goravel/framework/contracts/config"
+)
+
+// TimeoutMiddleware creates middleware to set a timeout for a request
+func TimeoutMiddleware() contractshttp.Middleware {
+	return func(ctx contractshttp.Context) {
+		timeout := config.GetInt("http.http.drivers.gin.timeout", 3) * time.Second
+		timeoutCtx, cancel := context.WithTimeout(ctx.Context(), timeout)
+		defer cancel()
+
+		// Refreshing the request context with a timeout
+		ctx.WithContext(timeoutCtx)
+
+		// We start executing the request in a new goroutine
+		done := make(chan struct{})
+		go func() {
+			defer close(done)
+			ctx.Request().Next()
+		}()
+
+		select {
+		case <-done:
```

Comment:
And modify this with above simultaneously.

```suggestion
		case <-ctx.Request().Origin().Context().Done():
```

Reply:
of course, so I need to remove the select block?

Reply:
No, as you tested, only setting context is not enough, `select` is required. I mean changing `done` to `ctx.Request().Origin().Context().Done()`, test it, check if it's correct.

Reply:
I tested it, it worked, I'll make changes today